crates/common/tedge_config/src/tedge_toml/tedge_config/mqtt_config.rs

```diff
@@ -142,34 +153,30 @@ impl TEdgeConfig {
         Ok(mqtt_config)
     }
 
-    /// Returns an authentication configuration for an MQTT client that will connect to the Cumulocity MQTT broker.
+    /// Returns an authentication configuration for an MQTT client that will connect to the MQTT
+    /// broker of a cloud given in the parameter.
     pub fn mqtt_auth_config_cloud_broker(
         &self,
-        c8y_profile: Option<&str>,
+        cloud: &dyn CloudConfig,
     ) -> anyhow::Result<MqttAuthConfigCloudBroker> {
-        let c8y = self.c8y.try_get(c8y_profile)?;
-
-        let mut mqtt_auth = MqttAuthConfigCloudBroker {
-            ca_dir: Some(c8y.root_cert_path.clone().into()),
-            ca_file: None,
-            client: None,
-        };
+        let ca_path = cloud.root_cert_path().to_path_buf();
 
         // if client cert is set, then either cryptoki or key file must be set
-        if let Some(cryptoki_config) = self.device.cryptoki.config()? {
-            mqtt_auth.client = Some(MqttAuthClientConfigCloudBroker {
-                cert_file: c8y.device.cert_path.clone().into(),
+        let client_auth = match self.device.cryptoki.config()? {
+            Some(cryptoki_config) => MqttAuthClientConfigCloudBroker {
+                cert_file: cloud.device_cert_path().to_path_buf(),
```

Comment:
`cloud.device_cert_path` can be either a file or path to a directory. Does this still work as expected?

Reply:
AFAIK `device_cert_path` the path is to a device certificate, so it can only be a file. It's derived from `device/c8y/aws/az.cert_path`, which point to files.

Reply:
Sorry my mistake. I was mixing this up with the ca_path...you're right this is just about the device certificate which is a file.